src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java

```diff
@@ -0,0 +1,115 @@
+import java.util.zip.ZipFile;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.yaml.snakeyaml.error.YAMLException;
+
+/**
+ * Test case for {@link AbstractBeanstalkMojo}.
+ * @author Neo Matrix (Neo.matrix@gmail.com)
+ * @version $Id$
+ */
+public final class GenericMojoTest {
+    /**
+     * {@link AbstractBeanstalkMojo} can execute successfully.
+     * @throws Exception If something is wrong.
+     * @checkstyle ExecutableStatementCountCheck (40 lines)
+     */
+    @Test
+    public void executesSuccessfully() throws Exception {
+        // @checkstyle IllegalTypeCheck (2 lines)
+        final AbstractBeanstalkMojo mojo =
+            Mockito.mock(AbstractBeanstalkMojo.class);
+        final File mockFile = Mockito.mock(File.class);
+        final ZipFile mockZipFile = Mockito.mock(ZipFile.class);
+        Mockito.when(mockFile.exists()).thenReturn(true);
+        Mockito.when(mojo.createZipFile()).thenReturn(mockZipFile);
+        Mockito.doCallRealMethod().when(mojo).execute();
+        Mockito.doCallRealMethod().when(mojo).getLog();
+        Mockito.doCallRealMethod().when(mojo)
+            .setWar(Mockito.any(File.class));
+        Mockito.doCallRealMethod().when(mojo)
+            .setName(Mockito.any(String.class));
+        Mockito.doCallRealMethod().when(mojo)
+            .setBucket(Mockito.any(String.class));
+        Mockito.doCallRealMethod().when(mojo)
+            .setKey(Mockito.any(String.class));
+        mojo.setWar(mockFile);
+        mojo.setName("name");
+        mojo.setBucket("bucket");
+        mojo.setKey("key");
+        mojo.execute();
+        Mockito.verify(mojo).createZipFile();
+        Mockito.verify(mojo).validate(mockZipFile);
+        Mockito.verify(mojo).createWarFile(mockZipFile);
+        Mockito.verify(mojo).validateWarFile(Mockito.any(WarFile.class));
+        Mockito.verify(mojo).createServerCredentials();
+        Mockito.verify(mockZipFile).close();
+        Mockito.verify(mockFile, Mockito.times(2)).exists();
+    }
+
+    /**
+     * {@link AbstractBeanstalkMojo} can validate bad yaml.
+     * @throws Exception If something is wrong
+     */
+    @Test(expected = YAMLException.class)
+    public void validatesBadYaml() throws Exception {
+        final String invalid = new StringBuilder()
+            .append("Some illegal Prefix\n")
+            .append("Time: 2005-11-23 10:01:42 -5\n")
+            .append("Admin: ed\n")
+            .append("Messages:\n")
+            .append(" Hello is an error information\n")
+            .append(" for the configuration file\n").toString();
+        final DeployMojo mojo = new DeployMojo();
+        mojo.validYaml(invalid);
+    }
+
+    /**
+     * {@link AbstractBeanstalkMojo} can validate good yaml.
+     * @throws Exception If something is wrong
+     */
+    @Test
+    public void validatesGoodYaml() throws Exception {
+        // @checkstyle StringLiteralsConcatenationCheck (6 lines)
+        final String valid = new StringBuilder()
```

Comment:
@tmjee this variable is redundant, see http://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html